core-blocks/latest-comments/index.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Server-side rendering of the `core/latest-comments` block.
+ *
+ * @package gutenberg
+ */
+
+$DEFAULT_COMMENTS_TO_SHOW = 5;
+
+/**
+ * Renders the `core/latest-comments` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the post content with latest comments added.
+ */
+function gutenberg_render_block_core_latest_comments( $attributes = array() ) {
+	// Basic attribute validation.
+	if (
+		! isset( $attributes['align'] ) ||
+		! in_array( $attributes['align'], array( 'center', 'left', 'right', 'wide', 'full' ), true )
+	) {
+		$attributes['align'] = null;
+	}
+	if (
+		! is_numeric( $attributes['commentsToShow'] ) ||
+		$attributes['commentsToShow'] < 0 ||
+		$attributes['commentsToShow'] > 100
+	) {
+		$attributes['commentsToShow'] = $DEFAULT_COMMENTS_TO_SHOW;
+	}
+
+	/** This filter is documented in wp-includes/widgets/class-wp-widget-recent-comments.php */
+	$comments = get_comments( apply_filters( 'widget_comments_args', array(
+		'number'      => $attributes['commentsToShow'],
+		'status'      => 'approve',
+		'post_status' => 'publish',
+	) ) );
+
+	$list_items_markup = '';
+	if ( ! empty( $comments ) ) {
+		// Prime cache for associated posts. This is copied from \WP_Widget_Recent_Comments::widget().
+		$post_ids = array_unique( wp_list_pluck( $comments, 'comment_post_ID' ) );
+		_prime_post_caches( $post_ids, strpos( get_option( 'permalink_structure' ), '%category%' ), false );
+
+		foreach ( $comments as $comment ) {
+			$list_items_markup .= '<li class="latestcomments">';
+			if ( $attributes['displayAvatar'] ) {
+				$avatar = get_avatar( $comment, 48, '', '', array(
+					'class' => 'wp-block-latest-comments__comment-avatar',
+				) );
+				if ( $avatar ) {
+					$list_items_markup .= $avatar;
+				}
+			}
+
+			$list_items_markup .= '<div class="comment-data"';
+			$author_url         = get_comment_author_url( $comment );
+			if ( empty( $author_url ) && ! empty( $comment->user_id ) ) {
+				$author_url = get_author_posts_url( $comment->user_id );
+			}
+			if ( $author_url ) {
+				$list_items_markup .= '<a class="wp-block-latest-comments__comment-author" href="' . esc_url( $author_url ) . '">' . get_comment_author( $comment ) . '</a>';
+			} else {
+				$list_items_markup .= '<a class="wp-block-latest-comments__comment-author">' . get_comment_author( $comment ) . '</a>';
+			}
+
+			$list_items_markup .= __( ' on ', 'gutenberg' );
+			$list_items_markup .= '<a class="wp-block-latest-comments__comment-link" href="' . esc_url( get_comment_link( $comment ) ) . '">' . get_the_title( $comment->comment_post_ID ) . '</a>';
```

Comment:
Just checked into this and it looks like `get_the_title()` doesn't do any escaping, so this is Dangerous™. I'll have it fixed in a branch and add tests for it.